diff -ru facter-1.5.8.orig/lib/facter/ec2.rb facter-1.5.8/lib/facter/ec2.rb
--- facter-1.5.8.orig/lib/facter/ec2.rb	2011-02-25 04:57:00.000000000 -0500
+++ facter-1.5.8/lib/facter/ec2.rb	2011-02-25 05:04:09.000000000 -0500
@@ -6,8 +6,11 @@
 require 'timeout'
 
 def can_connect?(ip,port,wait_sec=2)
- Timeout::timeout(wait_sec) {open(ip, port)}
+ url = "http://#{ip}:#{port}/";
+ Timeout::timeout(wait_sec) {open(url)}
  return true
+rescue Timeout::Error
+  return false
 rescue
   return false
 end
